File: Randy-JRA/Exercise_44.cpp
//Challenge 44 is to Create a function that will convert a string
//containing a binary number into a number

#include <iostream>
using namespace std;
// Function to convert binary to decimal
int binaryToDecimal(int n)
{
	int num = n;
	int dec_value = 0;

	// Initializing base value to 1, i.e 2^0
	int base = 1;

	int temp = num;
	while (temp) {
		int last_digit = temp % 10;
		temp = temp / 10;

		dec_value += last_digit * base;

		base = base * 2;
	}

	return dec_value;
}

int main()
{
	int num = 10101001;

	cout << binaryToDecimal(num) << endl;
}

File: AwembomN/Exercise44.py
'''Create a function that will convert a string containing a binary number into a
number'''

def binary_decimal():
    number = input("Enter the binary number you want to convert to dicimal: ")
    decimal = 0
    #j starts from minus so as to slice the string from behind
    j = -1
    for i in range(len(number)):
        value = int(number[j]) * pow(2, i)
        j -= 1
        decimal += value
    return decimal

answer = binary_decimal()
print(answer)